instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/package.py

```diff
@@ -12,6 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-_instruments = ("sqlalchemy",)
+_instruments = ("sqlalchemy >= 1.0.0, < 2.0.0",)
```

Comment:
Could we make the upper bound a bit less strict? The latest [release](https://pypi.org/project/SQLAlchemy/#history) is 2.0.36, so maybe something like

```suggestion
_instruments = ("sqlalchemy >= 1.0.0, < 2.1.0",)
```

We _could_ be more strict since [1.3.0 is eol](https://github.com/sqlalchemy/alembic/issues/1395), but maybe in another ticket:

```suggestion
_instruments = ("sqlalchemy >= 1.4.0, < 2.1.0",)
```

Reply:
@tammy-baylis-swi is the instrumentation working with 2.0?

Reply:
I tested with 2.0.35 and tests are failing (at least locally on my laptop)

Reply:
Yes, I can instrument a test app that uses SQLAlchemy 2.0.36 (with psycopg2 driver), latest Otel releases, and exports spans to collector.

Installed deps (Dockerfile):
```
RUN pip install \
    flask \
    requests \
    psycopg2 \
    sqlalchemy~=2.0.0

RUN pip install \
    opentelemetry-api \
    opentelemetry-sdk \
    opentelemetry-exporter-otlp \
    opentelemetry-distro \
    opentelemetry-instrumentation-dbapi \
    opentelemetry-instrumentation-psycopg2 \
    opentelemetry-instrumentation-sqlalchemy
```

Example span:
```
InstrumentationScope opentelemetry.instrumentation.sqlalchemy 0.49b0
...
Span #1
    Trace ID       : 36559251344aa8f8b9cca915d20628c4
    Parent ID      : e1367c7c11ea2213
    ID             : 2db48166c9b6779d
    Name           : SELECT world-db
    Kind           : Client
    Start time     : 2024-11-07 18:47:33.316962633 +0000 UTC
    End time       : 2024-11-07 18:47:33.467587216 +0000 UTC
    Status code    : Unset
    Status message :
Attributes:
     -> db.statement: Str(SELECT city.id, city.name, city.district, city.population, city.country_code
FROM city
WHERE city.id = %(id_1)s)
     -> db.system: Str(postgresql)
     -> net.peer.name: Str(34.238.40.181)
     -> net.peer.port: Int(5432)
     -> db.name: Str(world-db)
     -> db.user: Str(world)
```

Reply:
I have a CI fix in https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2976 which should allow for upper bound `< 2.1.0`

Reply:
Updating to < 2.1.0